cli/src/commands/parse.rs

```diff
@@ -33,46 +32,33 @@ pub fn handle_parse(matches: &clap::ArgMatches) -> CommandResult {
     // LOCKFILE is a required parameter, so .unwrap() should be safe.
     let lockfile = matches.value_of("LOCKFILE").unwrap();
 
-    let parser = LOCKFILE_PARSERS
-        .iter()
-        .filter_map(|(name, parser)| (*name == lockfile_type).then(|| parser))
-        .next()
-        .unwrap();
-
-    let (pkgs, _) = parser(Path::new(lockfile))?;
+    let (pkgs, _) = if lockfile_type == "auto" {
+        get_packages_from_lockfile(Path::new(lockfile))?
+    } else {
+        let parser = LOCKFILE_PARSERS
+            .iter()
+            .filter_map(|(name, parser)| (*name == lockfile_type).then(|| parser))
+            .next()
+            .unwrap();
+
+        let lockfile_content = fs::read_to_string(lockfile)?;
+        parser(lockfile_content)?
+    };
 
     serde_json::to_writer_pretty(&mut io::stdout(), &pkgs)?;
 
     Ok(ExitCode::Ok.into())
 }
 /// Attempt to get packages from an unknown lockfile type
-pub fn try_get_packages(path: &Path) -> Result<(Vec<PackageDescriptor>, PackageType)> {
-    log::warn!(
-        "Attempting to obtain packages from unrecognized lockfile type: {}",
-        path.to_string_lossy()
-    );
-
-    // Try a package lock format and return the packages if there are some.
-    macro_rules! try_format {
-        ($lock:ident, $ty:literal) => {{
-            let packages = parse::<$lock>(path).ok();
-            if let Some(packages) = packages.filter(|(packages, _)| !packages.is_empty()) {
-                log::debug!("Submitting file as type {}", $ty);
-                return Ok(packages);
-            }
-        }};
+pub fn try_get_packages(lockfile_content: String) -> Result<(Vec<PackageDescriptor>, PackageType)> {
+    for (ty, parser) in LOCKFILE_PARSERS.iter().filter(|(ty, _)| ty != &"auto") {
```

Comment:
There's no need to call `.filter()` here now that `auto` has been removed from the list of parsers.